internal/s3storage/s3storage.go

```diff
@@ -1,54 +1,149 @@
 package s3storage
 
 import (
+	"fmt"
 	"time"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/awserr"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/sqs"
 	"github.com/lunarway/release-manager/internal/artifact"
 	"github.com/lunarway/release-manager/internal/log"
 	"github.com/lunarway/release-manager/internal/tracing"
 	"github.com/pkg/errors"
 )
 
 type Service struct {
-	bucketName string
-	s3client   *s3.S3
-	tracer     tracing.Tracer
+	region                 string
+	bucketName             string
+	s3client               *s3.S3
+	sqsClient              *sqs.SQS
+	tracer                 tracing.Tracer
+	sqsQueueURL            string
+	sqsQueueARN            string
+	sqsHandlerQuitChannel  chan struct{}
+	sqsHandlerErrorChannel chan error
 }
 
 func New(bucketName string, tracer tracing.Tracer) (*Service, error) {
+	region := "eu-west-1"
 	sess, err := session.NewSession(&aws.Config{
-		Region: aws.String("eu-west-1"),
+		Region: aws.String(region),
 	})
 	if err != nil {
 		return nil, err
 	}
-	// Create a S3 client from just a session.
+
 	s3client := s3.New(sess)
+	sqsClient := sqs.New(sess)
 
 	return &Service{
+		region:     region,
 		bucketName: bucketName,
 		s3client:   s3client,
+		sqsClient:  sqsClient,
 		tracer:     tracer,
 	}, nil
 }
 
+// InitializeSQS set up a subscription from S3 to a SQS queue. The SQS queue events triggers calls to handler.InitializeSQS
+//
+// Note on S3 notifications:
+// There are 2 ways to get S3 notifications into SQS:
+//  - Directly from S3 Notification to SQS Queue
+//  - From S3 Notification to SNS Topic to SQS Queue
+// Using a SNS Topic should be more powerful, since it fx can send to multiple SQS Queues, but it requires more configuration
+// and moving parts. The simpler model should suffice, therefore we connect it directly.k
+func (s *Service) InitializeSQS(handler func(msg string) error) error {
+	// Amazon SQS returns only an error if the request includes attributes whose values differ from those of the existing queue.
+	queue, err := s.sqsClient.CreateQueue(&sqs.CreateQueueInput{
+		QueueName: aws.String(s.sqsQueueName()),
+	})
+	if err != nil {
+		return errors.Wrap(err, "create sqs queue")
+	}
+	log.Infof("Created queue with url: %s", *queue.QueueUrl)
+	s.sqsQueueURL = *queue.QueueUrl
+
+	queueAttributes, err := s.sqsClient.GetQueueAttributes(&sqs.GetQueueAttributesInput{
+		QueueUrl:       queue.QueueUrl,
+		AttributeNames: []*string{aws.String(sqs.QueueAttributeNameQueueArn)},
+	})
+	if err != nil {
+		return errors.Wrap(err, "get sqs queue ARN")
+	}
+	s.sqsQueueARN = *queueAttributes.Attributes[sqs.QueueAttributeNameQueueArn]
+	log.Infof("SQS queue arn acquired: %s", s.sqsQueueARN)
+
+	policy := `{
+		"Version": "2012-10-17",
+		"Id": "` + s.sqsQueueARN + `/SQSDefaultPolicy",
+		"Statement": [
+		 {
+			"Sid": "` + s.bucketName + `-s3-permission",
```

Comment:
Just a sec note. `s.bucketName` is inserted here directly from a flag and thus can be used to escape this statement eg bucket name `my-bucket" }]}/*`

It might not be any concern of ours as this is a server startup flag and not dynamically setup.

Reply:
Yeah, don't think its too much of a concern, due to running release-manager is going through another review process in `lunar-way-release-manager`.
Also if you tried to use tried to use `my-bucket" }]}/*` I would guess the CreateBucket step before will fail hard 😅